worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/EmailSplitter.java

```diff
@@ -89,7 +91,10 @@ public void generateEmailTags(Document doc) throws JDOMException, ExecutionExcep
                     if (matcher.find()) {
                         divider = matcher.group(DIVIDER_GROUP_ID);//group 1 matches the divider in the regex above
                         if (dividerConfirmationPattern.matcher(divider).find()) {
-                            email = email.substring(0, email.indexOf(divider));
+                            final String emailTemp = email.substring(0, email.indexOf(divider));
+                            final String restOfEmail = email.substring(emailTemp.length() + divider.length());
+                            email = emailTemp;
+                            divider = restOfEmail.matches(dividerPatternString) ? divider + restOfEmail : divider;
```

Comment:
It doesn't make sense to compile the pattern all over again.  Could you not use the pattern you already compiled?

    dividerPattern.matcher(restOfEmail).matches()

Reply:
@dermot-hardy Updated to reflect comment